File: src/dartCode/search_in_range.dart
void main() {
  List<int> arr = [18, 12, -7, 3, 14, 28];
  int target = 12;
  print(linearSearch(arr, target, 0, 5));
}

int linearSearch(List<int> arr, int target, int start, int end) {
  if (arr.length == 0) {
    return -1;
  }

  for (int index = start; index <= end; index++) {
    int element = arr[index];
    if (element == target) {
      return index;
    }
  }
  return -1;
}

File: src/kotlinCode/SearchInRange.kt
package kotlinCode

fun main() {
    var arr = intArrayOf(18, 12, -7, 3, 14, 28)
    var target = 14
    println(linearSearch(arr, target, 0, arr.size))
}

// FIXME: 03/06/22  
fun linearSearch(arr: IntArray, target: Int, start: Int, end: Int): Int {
    if (arr.isEmpty()) {
        return -1;
    }
    var index = start
    (index..arr.size).takeWhile { start <= end }.forEach { _ ->
        var element = arr[index]
        if (element == target) {
        }
            return target
    }
    return -1
}